src/main/java/org/svydovets/session/Session.java

```diff
@@ -92,6 +94,72 @@ public <T> T findById(Class<T> entityType, Object id) {
         return entityType.cast(entity);
     }
 
+    /**
+     * Retrieves an entity by its class type and identifier from the cache or database.
+     *
+     * @param entityType The class of the entity to retrieve.
+     * @param field The entity field.
+     * @param columnValue The entity field value.
+     * @param <T> The type of the entity.
+     * @return The found entity or null if not found.
+     */
+    public <T> T findBy(final Class<T> entityType, final Field field, final Object columnValue) {
+        checkIfOpenSession();
+
+        T entity = jdbcDAO.findBy(entityType, field, columnValue);
+
+        return entityType.cast(computeIfAbsent(entity));

```

Comment:
I guess we have to check if entity is in the cache first and if it's not we will load from the DB by "jdbcDAO.findBy()", so I would rewrite this method "findBy()" and "computeIfAbsent()"